node.go

```diff
@@ -471,12 +471,14 @@ func (n *node) dereference() {
 	for i := range n.inodes {
 		inode := &n.inodes[i]
 
-		key := make([]byte, len(inode.Key()))
+		buf := make([]byte, len(inode.Key())+len(inode.Value()))
+
+		key := buf[:len(inode.Key())]
```

Comment:
Full slice expression `buf[:len(inode.Key()):len(inode.Key())]` restricts the capacity and prevents possible bugs, were we to append something to key.
What do you think?

Reply:
good point